### PR TITLE
m_len check in rsa_sign()

### LINKED ISSUE #14594: crypto/rsa/rsa_sign.c RSA_Sign() input isn't clear
RSA_Sign() expects a buffer containing the data to sign. It isn't clear this is the message digest and not the message itself. Inputting the message and not the message digest could cause information to leak. 

It would help developers to either change m to m_digest, or add a check to lines 299-303 to verify the length of m is correct for the hash function.

### PR DESCRIPTION
**Pull request by kylon94**
Technical check of the expected length of m_len in rsa_sign(). I have added a look up on the hash function id using EVP_get_digestbynid(), then a comparison of m_len against the result of a call to EVP_MD_size(). 

Fixes #14594

### CODE REVIEW

File: crypto/rsa/rsa_sign.c [hunk 1]
```diff
@@ -91,6 +91,17 @@ int RSA_sign(int type, const unsigned char *m, unsigned int m_len,
         encoded_len = SSL_SIG_LENGTH;
         encoded = m;
     } else {
+        const EVP_MD *md = EVP_get_digestbynid(type);
+        if (md == NULL) {
+            RSAerr(RSA_F_RSA_SIGN, RSA_R_UNKNOWN_ALGORITHM_TYPE);
+            return 0;
+        }
```
Comment (t8m) [inline crypto/rsa/rsa_sign.c:98]: This is breaking compatibility. If the type does not resolve to an md, the m_len check should be simply skipped. Also please add empty line after declaration.

**Comment by slontis** [comment]
Probably not worth doing this for master since the API is deprecated..

File: crypto/rsa/rsa_sign.c (hunk 1 above)
Comment (kylon94) [reply crypto/rsa/rsa_sign.c]: Just pushed a commit to address this.

**Comment by kylon94** [comment]
Closed (and reopening) commit to remove the "cla required" tag.

**Comment by t8m** [comment]
The cla check did not recognize the anonymized e-mail used in the commit.

**Pull request closed without merging**